plenum/server/consensus/ordering_service.py

```diff
@@ -2376,19 +2384,29 @@ def process_old_view_preprepare_request(self, msg: OldViewPrePrepareRequest, sen
         self._send(rep, dst=[getNodeName(sender)])
 
     def process_old_view_preprepare_reply(self, msg: OldViewPrePrepareReply, sender):
+        # TODO: return the check after INDY-2238 about persisting of 3pc messages
+        # At the moment this optimization adds a bug that is reproduced in
+        # test_view_change_with_delayed_commits_on_half_of_the_nodes_and_restart_of_the_other_half
+        # if self._data.prev_view_prepare_cert and self._data.prev_view_prepare_cert <= self.lastPrePrepareSeqNo:
+        #     self._reordered_in_new_view()
+        #     return
         result, reason = self._validate(msg)
         if result != PROCESS:
             return result, reason
 
         for pp_dict in msg.preprepares:
             try:
                 pp = PrePrepare(**pp_dict)
+                if self._data.new_view is None or \
+                        preprepare_to_batch_id(pp) not in self._data.new_view.batches:
+                    self._logger.info("Skipped useless PrePrepare {} from {}".format(pp, sender))
```

Comment:
I think this needs to be warning, and the  message need to indicate that a sender Node sent incorrect PrePrepare to us. This is an indication that either the sender node is malicious, or it has incorrect PrePrepares for some reasons.